docs/DeveloperGuide.md

```diff
@@ -1,29 +1,165 @@
+![img.png](images/Event_class.png)
+
+The following sequence diagrams show how a user input is processed to add the events: </br>
+
+![img.png](images/addEventCommand.png) </br>
+![img.png](images/executeAddEventCommand.png)
+
+**User input event main parsing sequence**
+1. The user enters a command to add an event
+2. The `EventParser` class parses the user input to identify the event name and the event description
+3. The `AddEventCommand` object is created with the event name and description
+4. When executed, the `AddEventCommand` object creates an `Event` object with the event name and description
+5. The `AddEventCommand` object calls the `addEvent` method in the `EventManager` class to add the event created in step 4
+6. Feedback is given to the user via the UI
+
+Details such as accessing the EventManager via the central dashboard and handling of exceptions using invalid commands are omitted for brevity.
+
+The following sequence diagram shows the execution of an `AddExpenseToEventCommand` command object </br>
+
+![img.png](images/executeAddExpenseToEventCommand.png)
+
+**Execution sequence of AddExpenseToEventCommand**
+1. The command checks the validity of the event and expense indexes 
+2. If the indexes are invalid, a feedback message is returned to the user
+3. The event and expense objects are retrieved based on the indexes
+4. If the expense already exists in the event, a feedback message is returned to the user
+5. The original owning event of the expense is retrieved 
+6. If the expense has an owning event, the expense is removed from the owning event
+7. The expense is added to the new event, and its owning event is updated
 
 ## Product scope
 ### Target user profile
 
-{Describe the target user profile}
+This project is designed to cater to the needs of university students who encounter the challenge of managing a myriad of expenses across various categories.
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+Brokeculator is a CLI application designed for university students to log and view their expenses. 
+For experienced CLI users, they can enter their expenses faster compared to GUI applications
 
 ## User Stories
 
 |Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| v1.0 | student | see a basic summary of my expenses to see how much i have spent in total | ------------------ |
+| v1.0 | student | view the expenses I have logged | know how much I have spent |
+| v1.0 | paranoid user | save my expenses into a file | backup locally via a file to prevent data loss |
+| v1.0 | student | have the ability to add expenses | ------- |
```

Comment:
Perhaps you could make the user stories more complete by filling in the 'so that I can' part.